### PR TITLE
handle unattached hardware reservations

### PR DESCRIPTION
**Pull request by boz**
This is currently failing with:

```
TypeError: 'NoneType' object is not subscriptable
```

### CODE REVIEW

File: packet/HardwareReservation.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ def __init__(self, data, manager):
         # except (KeyError, IndexError):
         #     self.attached_to = None
 
+
+        # not attached
+        if data["device"] is None:
+            self.attached_to = None
```
Comment (displague) [inline packet/HardwareReservation.py:50]: Alternatively, we could use `except (KeyError, IndexError, TypeError):` in the following `try` block.

Comment (displague) [reply packet/HardwareReservation.py:50]: @boz I think I would prefer that approach.  The overfit error handling is mildly concerning, but I don't see any risks that are not already present in the previous `try` block.

Comment (boz) [reply packet/HardwareReservation.py:50]: Sure, will do.

I it that way at first but I felt like it could shadow other problems too easily.

**Review by displague — APPROVED** [review]

**Comment by boz** [comment]
Thanks so much @displague - sorry I ran out of bandwidth, was just coming back to this.

**Pull request merged**